lib/widget/button.dart

```diff
@@ -89,25 +92,37 @@ class ButtonState extends WidgetState<Button> {
     bool isOutlineButton = widget._controller.outline ?? false;
 
     ScopeManager? scopeManager = DataScopeWidget.getScope(context);
-    Widget label;
-    if (scopeManager != null && widget._controller.labelWidget != null) {
-      label = scopeManager.buildWidgetWithScopeFromDefinition(widget._controller.labelWidget);
-    } else {
-      label = Text(
-          Utils.translate(widget._controller.label ?? '', context));
+    Widget? startingIcon;
+    Widget? endingIcon;
+    if (widget._controller.startingIcon != null) {
+      startingIcon = scopeManager?.buildWidgetFromDefinition(
+          widget._controller.startingIcon);
+    }
+    if (widget._controller.endingIcon != null) {
+      endingIcon = scopeManager?.buildWidgetFromDefinition(widget._controller.endingIcon);
+    }
+    List<Widget> labelParts = [Text(Utils.translate(widget._controller.label ?? '', context))];
+    if (startingIcon != null) {
+      labelParts.insertAll(0, [startingIcon]);
```

Comment:
why not just use `add(endingIcon)` and `insert(0, startingIcon)`?

Reply:
This is a holdover from when I was going to add gaps in between the icon and label.